src/python/RestClient/AuthHandling/X509Auth.py

```diff
@@ -82,16 +88,28 @@ def __search_cert_key(self):
         else:
             raise ClientAuthException("No valid X509 cert-key-pair found.")
 
+    def __create_ca_bundle(self):
+        trust_ca_files = glob.glob(f"{self._ca_path}/*.pem")
+        trust_ca_files.append(self._ssl_cert)
+
+        X509Auth.bundles = NamedTemporaryFile()

```

Comment:
I'm still uncomfortable with the approach of using file to keep CAs. I did not check explicitly but the tempfile will be deleted as soon as it [closed](https://docs.python.org/2/library/tempfile.html#tempfile.NamedTemporaryFile). That's fine but I don't recall if it will work in 100% of the cases when your program can be killed from outside.

Said that, what you actually need is [file-like object](https://docs.python.org/3/glossary.html#term-file-object), i.e. an object with `read()` and `write()` method to underlying resources. Therefore, you can make a class which will provide these methods and keep your data in memory, e.g.
```
class CABundles():
    def __init__(self):
        self.data = None
    def read(self):
        return self.data
    def write(self, data):
        self.data = data
```
Then, you'll use it instead of tempfile and your code will never leak anything to file system and bundels will be kept in memory. But you'll use the same logic to read it once and provide such file-like object to curl lib.

Reply:
@vkuznet As far as I know, the pycurl library is known as the Python binding of libcurl, and the corresponding one to the CAINFO option of the pycurl is the CURLOPT_CAINFO of libcurl. 

The libcurl library receives ```char*``` type of path information as an argument(https://curl.se/libcurl/c/CURLOPT_CAINFO.html), so I think it must give an existing file. From curl 7.77.0, a variable called CURLOPT_CAINFO_BLOB appears to be able to make temporary files only exist in memory as you wish. However, the curl version provided by CentOS7 is 7.29, so it seems difficult to use. 

I thought it wouldn't be a big problem if the CA bundle file was leaked because it was a bunch of (opened-)public keys. 

However, if the method appears to be problematic, it doesn't matter if you reject it because there are many solutions, such as abandoning the CentOS7 native environment and compiling libcurl separately(openssl backend) or storing user certificates by a certutil. I was going to update it so that I can use it comfortably.

Reply:
If this PR is rejected, I am going to upload a new PR so that only ca_info can be used as I requested in the beginning. I think it's a safer way to do that version because it does not create a temporary CA bundle in the X509Auth class and users use the ca_info which is created by themselves.

Reply:
@geonmo , you're right about `char*` type, the setopt for CAINFO expects a file name. So, in this case NamedTemporaryFile should be sufficient. I'm ok with that.

@amaltaro do you have any other opinion about it, if not we can proceed with this PR. But I do not know how to specifically perform tests you are asking for in unittest. We basically need to pick up dedicated nodes and run the code over there. @geonmo can you perform these tests on node flavors Alan asked for?

Reply:
@vkuznet I used dbsClient code as below.
```python
#!/usr/bin/env python3
import os
import socket
from RestClient.ErrorHandling.RestClientExceptions import HTTPError
from RestClient.RestApi import RestApi
from RestClient.AuthHandling.X509Auth import X509Auth
datasetname = "/TTTo2L2Nu_hdampDOWN_TuneCP5_13TeV-powheg-pythia8/RunIISummer20UL18NanoAODv9-106X_upgrade2018_realistic_v16_L1v1-v1/NANOAODSIM"
instance = "global"
url = f"https://cmsweb.cern.ch/dbs/prod/{instance}/DBSReader"
if url.find(":", 6) == -1:
    url = url.replace(".cern.ch/dbs/", ".cern.ch:" + "8443" + "/dbs/", 1)
# Default
cert=None
key=None
verifypeer= True
ca_info = None
# Set proxyCA
#cert=f"/tmp/x509up_u{os.getuid()}"
#key=cert
# Set ca_info manually
#ca_info = "cert.crl"
rest_api = RestApi(auth=X509Auth(ssl_cert=cert, ssl_key=key, ssl_verifypeer=verifypeer, ca_info=ca_info), proxy=None)
method_func = getattr(rest_api, "get")
params={'dataset': datasetname, 'detail': 1}
headers={'Content-Type': 'application/json', 'Accept': 'application/json', 'UserID': f'{os.environ["USER"]}@{socket.gethostname()}', 'User-Agent': 'DBSClient/Unknown/'}
response = method_func(url, "datasets", params, None, headers)
print(response.body)
print(headers)
```

Reply:
thanks, the code looks fine for me, and I'm ok with this PR merge. Let gets @amaltaro approval too.